src/main/java/spiffe/api/svid/SpiffeEndpointChannelBuilder.java

```diff
@@ -0,0 +1,117 @@
+ *  Based on the Address scheme, creates a TCP or a Native Socket Channel
+ *  Configures the Socket Channel based on the detected OS
+ *
+ */
+class SpiffeEndpointChannelBuilder {
+
+    static final String ENV_ADDRESS_VAR = "SPIFFE_ENDPOINT_SOCKET";
+
+    /**
+     * Returns a configured ManagedChannel
+     * @param spiffeEndpointAddress
+     * @return
+     */
+    static ManagedChannel newChannel(String spiffeEndpointAddress) {
+        if (isBlank(spiffeEndpointAddress)) {
+            spiffeEndpointAddress = getAddressFromEnv();
+        }
+
+        URI parsedAddress = parseAddress(spiffeEndpointAddress);
+        if (isTcp(parsedAddress)) {
+            return createTcpChannel(parsedAddress);
+        } else {
+            return createNativeSocketChannel(spiffeEndpointAddress);
+        }
+    }
+
+    /**
+     * Try to resolve the Address from the Environment.
+     * @throws IllegalStateException if the Address is not found
+     * @return the Address
+     */
+    private static String getAddressFromEnv() {
+        String address = System.getenv(ENV_ADDRESS_VAR);
+        if (isBlank(address)) {
+            throw new IllegalStateException(ENV_ADDRESS_VAR + " env var is not defined");
+        }
+        return address;
+    }
+
+    /**
+     * Create a TCP channel based on the given URI
+     *
+     * @param parsedAddress
+     * @return
+     */
+    private static ManagedChannel createTcpChannel(URI parsedAddress) {
+        return NettyChannelBuilder.forAddress(parsedAddress.getHost(), parsedAddress.getPort())
+                .negotiationType(NegotiationType.PLAINTEXT)
+                .build();
+    }
+
+    /**
+     * Create a Native Socket Channel pointing to the spiffeEndpointAddress
+     * @param spiffeEndpointAddress
+     * @return
+     */
+    private static ManagedChannel createNativeSocketChannel(String spiffeEndpointAddress) {
+        NettyChannelBuilder channelBuilder = NettyChannelBuilder.
+                forAddress(new DomainSocketAddress(spiffeEndpointAddress));
+        configureNativeSocketChannel(channelBuilder);
+        return channelBuilder
+                .usePlaintext()
+                .build();
+    }
+
+    /**
+     * Based on the detected OS, configures the Socket Channel EventLookGroup and Channel Type
+     *
+     * @param channelBuilder
+     */
+    private static void configureNativeSocketChannel(NettyChannelBuilder channelBuilder) {
+        if (SystemUtils.IS_OS_LINUX) {
+            channelBuilder.eventLoopGroup(new EpollEventLoopGroup())
+                          .channelType(EpollDomainSocketChannel.class);
+        } else if (SystemUtils.IS_OS_MAC) {
+            channelBuilder.eventLoopGroup(new KQueueEventLoopGroup())
+                          .channelType(KQueueServerDomainSocketChannel.class);
+        } else {
+            channelBuilder.eventLoopGroup(new NioEventLoopGroup());
+        }
+    }
+
+    private static URI parseAddress(String spiffeEndpointAddress) {
```

Comment:
We may want to have some validations like we do here: https://github.com/spiffe/spire/blob/master/api/workload/dial/dial.go#L78

Reply:
I added the validations.

Reply:
I think UrlValidator will not be of use to validate that the path is absolute, for example.